messages/java/src/main/java/io/cucumber/messages/MessageToNdjsonWriter.java

```diff
@@ -15,7 +15,7 @@ public MessageToNdjsonWriter(OutputStream out) {
 
     @Override
     public void write(Object message) throws IOException {
-        JSON.writeValue(out, message);
+        Jackson.OBJECT_MAPPER.writeValue(out, message);
```

Comment:
I wonder, could we remove Jackson and implement this in Cucumber JVM?

Maybe construct this class with a `Function<Object, String> serializer` and only use Jackson in tests.

Reply:
Ah wait. Wrong function signature.

Reply:
Sure, that should be doable. Let me handle that in a separate PR.

Reply:
`MessageToNdjsonWriter` should also implement closable.

Reply:
What about the deserializer we'd have to add to `NdjsonToMessageIterable`? `BiFunction`?

Reply:
Do we ever de-serialize outside of the tests?

Reply:
No production code uses deserialization, but we do use it in Gherkin's acceptance tests.

I think it would be nice to ship a general `NdjsonToMessageIterable` class with the library, but force users to construct it with a deserializer.

If we refactor the Gherkin acceptance tests to be parameterised tests a la cucumber expressions, we can do it all in JUnit, and we wouldn't need to depend on Jackson in Gherkin. Alternatively we move the Gherkin `Main` class over to `test`.

Reply:
Ok. Sounds like a separate MR.